sklearn/ensemble/_gb.py

```diff
@@ -64,11 +64,12 @@
 
 def _safe_divide(numerator, denominator):
     """Prevents overflow and division by zero."""
-    with np.errstate(divide="raise"):
-        try:
-            return numerator / denominator
-        except FloatingPointError:
-            return 0.0
+    # We could implement the following using `np.errstate` but, at the time of
+    # writing (scikit-learn 1.4), `np.errstate` is not handled properly by Pyodide.
+    # We therefore risk to issue `nan` values, making GBDT training fail.
+    if abs(denominator) < 1e-150:
+        return 0.0
```

Comment:
I really do not like such arbitrary cutoffs. For instance, the example where this happens uses nonsense params: learning_rate=1, subsample=0.5 with n_samples=800. (However, good settings for demonstration).
How about:
```python
return float(numerator) / float(denominator)
```

Reply:
If we need a cutoff-like solution, I would prefer to have them in the if clauses of the losses.